src/main/java/org/mycore/externalstore/rest/MCRExternalStoreResource.java

```diff
@@ -0,0 +1,301 @@
+
+    private static final String PARAM_STORE_TYPE = "store_type";
+
+    private static final String PARAM_PATH = "path";
+
+    private static final String PARAM_DOWNLOAD_TOKEN = "download_token";
+
+    private static final String CREATE_DERIVATE_PERMISSION = "create-derivate";
+
+    private static final String TOKEN_DOWNLOAD_AUDIENCE = "mcr:es:download";
+
+    private static final MCRExternalStoreInfoIndex INDEX = MCRExternalStoreInfoIndexManager.getInfoIndex();
+
+    @GET
+    @Path("{" + PARAM_OBJ_ID + "}/info")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response listInfo(@PathParam(PARAM_OBJ_ID) MCRObjectID objectId) {
+        ensureObjectExists(objectId);
+        if (!MCRAccessManager.checkPermission(objectId, MCRAccessManager.PERMISSION_READ)) {
+            throw new ForbiddenException();
+        }
+        final List<MCRDerivateInfoDto> derivateInfos
+            = MCRExternalStoreResourceHelper.listDerivateInformations(objectId);
+        final boolean canCreateStore = checkCreateStorePermission(objectId.toString());
+        final MCRDerivateInfosDto result = new MCRDerivateInfosDto(derivateInfos, canCreateStore);
+        return Response.ok(result).header("Access-Control-Allow-Origin", "*").build();
+    }
+
+    @POST
+    @Path("{" + PARAM_OBJ_ID + "}/add/{" + PARAM_STORE_TYPE + "}/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @MCRRequireTransaction
+    public Response createStore(@PathParam(PARAM_OBJ_ID) MCRObjectID objectId,
+        @PathParam(PARAM_STORE_TYPE) String storeType, Map<String, String> storeProviderSettings) {
+        ensureObjectExists(objectId);
+        if (!checkCreateStorePermission(objectId.toString())) {
+            throw new ForbiddenException();
+        }
+        try {
+            MCRExternalStoreService.createStore(objectId, storeType, storeProviderSettings);
+        } catch (MCRExternalStoreException e) {
+            throw new BadRequestException(e);
+        }
+        return Response.ok().header("Access-Control-Allow-Origin", "*").build();
+    }
+
+    @GET
+    @Path("{" + PARAM_OBJ_ID + "}/list/{" + PARAM_DER_ID + "}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response listFilesInfos(@PathParam(PARAM_OBJ_ID) MCRObjectID objectId,
+        @PathParam(PARAM_DER_ID) String base64DerivateId, @DefaultValue("0") @QueryParam("offset") int offset,
+        @DefaultValue("2147483647") @QueryParam("limit") int limit) {
+        return listFilesInfos(objectId, base64DerivateId, "", offset, limit);
+    }
+
+    @GET
+    @Path("{" + PARAM_OBJ_ID + "}/list/{" + PARAM_DER_ID + "}/{" + PARAM_PATH + "}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response listFilesInfos(@PathParam(PARAM_OBJ_ID) MCRObjectID objectId,
+        @PathParam(PARAM_DER_ID) String base64DerivateId, @PathParam(PARAM_PATH) String base64Path,
+        @DefaultValue("0") @QueryParam("offset") int offset,
+        @DefaultValue("2147483647") @QueryParam("limit") int limit) {
```

Comment:
I think in this case instead of 2147483647, you should use ""+Integer.MAX_VALUE, or make a constant with a name.